src/plugins/intel_npu/src/al/src/config/config.cpp

```diff
@@ -244,6 +244,31 @@ std::string Config::toString() const {
     return resultStream.str();
 }
 
+void Config::fromString(const std::string& str) {
+    std::map<std::string, std::string> config;
+    std::string str_cfg(str);
+
+    auto parse_token = [&](const std::string& token) {
+        auto pos_eq = token.find('=');
+        auto key = token.substr(0, pos_eq);
+        auto value = token.substr(pos_eq + 2, token.size() - pos_eq - 3);
+        config[key] = value;
+    };
+
+    size_t pos = 0;
+    std::string token, key, value;
+    while ((pos = str_cfg.find(' ')) != std::string::npos) {
+        token = str_cfg.substr(0, pos);
+        parse_token(token);
+        str_cfg.erase(0, pos + 1);
+    }
+
+    // Process tail
+    parse_token(str_cfg);
+
+    update(config);
+}
```

Comment:
Technically we need @PatrikStepan to look at this, but as it is _new_ code, never used before -- shouldn't break anything. In the worst case we'll have time on Monday to address feedback, so let's have this bulk of code merged today

Reply:
THIS code certainly needs tests as it doesn't handle any corner cases & errors.